src/Simplex/Messaging/Agent.hs

```diff
@@ -1762,6 +1755,13 @@ prepareDeleteConnections_ getConnections c waitDelivery connIds = do
     rcvQueues (SomeConn _ conn) = case connRcvQueues conn of
       [] -> Left $ Right ()
       rqs -> Right rqs
+    unsubConnIds :: NonEmpty ConnId -> AM' ()
+    unsubConnIds connIds' = do
+      forM_ connIds' $ \connId ->
+        -- TODO [batch ntf] remove as list / all atomically?
```

Comment:
no, better multiple transactions

Reply:
or you mean single map update? yes, it will be faster. I thought you mean moving atomically outside of the loop